src/parser/func_type.hpp

```diff
@@ -98,16 +98,20 @@ namespace Func
     INCOMPATIBLE
   };
 
-  struct argument_compatibility
+  struct argument_resolve
   {
-    type_compatibility m_compat;
     PT_TYPE_ENUM m_type;
+    pt_coll_infer m_coll_infer;
   };
 
   struct signature_compatibility
   {
-    type_compatibility m_signature_compat;
-    std::vector<argument_compatibility> m_args_compat;
+    type_compatibility m_compat;
+    std::vector<argument_resolve> m_args_resolve;
+    pt_coll_infer m_collation_inference;
```

Comment:
I would rename as 'm_expected_collation' or 'm_common_collation'.
This refers to the signature's common collation information.
With current naming, it could be confused with argument_resolve.m_coll_infer which refers to an actual argument.